server.go

```diff
@@ -172,6 +150,25 @@ func (s *Server) Close() error {
 	return err
 }
 
+func (s *Server) createAllocationManager(i int, addrGenerator RelayAddressGenerator, handler PermissionHandler) (*allocation.Manager, error) {
+	permissionHandler := handler
+	if permissionHandler == nil {
+		permissionHandler = DefaultPermissionHandler
+	}
+
+	allocationManager, err := allocation.NewManager(allocation.ManagerConfig{
+		AllocatePacketConn: addrGenerator.AllocatePacketConn,
+		AllocateConn:       addrGenerator.AllocateConn,
+		PermissionHandler:  permissionHandler,
+		LeveledLogger:      s.log,
+	})
+	if err != nil {
+		return allocationManager, err
+	}
+	s.allocationManagers[i] = allocationManager
```

Comment:
I am just wondering what happens if the user provides both `listenerConfigs` and `packetConnConfigs`?
Arn't those indices then overwritten by each other? Or are those two settings mutually exclusive?

Reply:
Oh, nevermind. I figured it out..